src/ngraph/op/experimental/range.cpp

```diff
@@ -240,8 +240,10 @@ void op::Range::validate_and_infer_types()
 #if defined(__GNUC__) && !(__GNUC__ == 4 && __GNUC_MINOR__ == 8)
 #pragma GCC diagnostic pop
 #endif
-
-    set_output_type(0, result_et, result_shape);
+    if (!result_shape.is_dynamic())
+    {
+        set_output_type(0, result_et, result_shape);
```

Comment:
dont we need to set the output shape to dynamic {} if it cannot infer out shape. Also, is this change needed for this feature , as we propagate in pass.

Reply:
We defer the shape inference to runtime if we dont run the max propagation pass. 
If we run the pass the output shape is being set to max from the pass

Reply:
Let's assume we always run the max propagation pass. No need to make things complicated for a pass that should be low-overhead if no maxes have been set.

Reply:
The problem is the max output shape gets overwritten by some other pass or elsewhere. we call the validate and infer a bunch of times from various places in the code.